library/spdm_requester_lib/libspdm_req_finish.c

```diff
@@ -370,15 +372,16 @@ static libspdm_return_t libspdm_try_send_receive_finish(libspdm_context_t *spdm_
 
     spdm_request->header.spdm_version = libspdm_get_connection_version (spdm_context);
     spdm_request->header.request_response_code = SPDM_FINISH;
+    spdm_request->header.param1 = 0;
+    spdm_request->header.param2 = 0;
+    signature_size = 0;
     if (session_info->mut_auth_requested) {
```

Comment:
There's nothing in `libspdm_req_key_exchange.c` that checks the Requester's capabilities, either `LIBSPDM_ENABLE_CAPABILITY_MUT_AUTH_CAP` or in the SPDM context, for mutual authentication. It just blindly sets `session_info->mut_auth_requested` to whatever the Responder tells it, even if it doesn't support mutual authentication. That should be fixed in this pull request.

Reply:
add check in req_key_exchange.